client/src/procedures/primitives.rs

```diff
@@ -484,20 +490,41 @@ impl DeriveSecret<1> for Slip10Derive {
     type Output = ChainCode;
 
     fn derive(self, guards: [Buffer<u8>; 1]) -> Result<Products<ChainCode>, FatalProcedureError> {
-        let dk = match self.input {
+        let (extended_bytes, chain_code) = match self.input {
             Slip10DeriveInput::Key(_) => {
                 let r = &*guards[0].borrow();
-                let ext_bytes: &[u8; 64] = r
+                let ext_bytes: &[u8; 65] = r
                     .try_into()
                     .map_err(|_| FatalProcedureError::from("bad slip10 extended secret key size".to_owned()))?;
-                slip10::ExtendedSecretKey::try_from_extended_bytes(self.curve, ext_bytes)
-                    .and_then(|parent| parent.derive(&self.chain))
+                match self.curve {
```

Comment:
Would it be too much to add a generic key type to the `Slip10Derive` rather than have `Curve`?

Reply:
Yes, actually. First, `hazmat` traits are inaccessible because `hazmat` module is private. Second, it's common for stronghold to have runtime types (see [`KeyType`](https://github.com/iotaledger/stronghold.rs/blob/snapshot-migration-v3age-2.0/client/src/procedures/primitives.rs#L325) and [`AeadCipher`](https://github.com/iotaledger/stronghold.rs/blob/snapshot-migration-v3age-2.0/client/src/procedures/primitives.rs#L319) enums, [`GenerateKey`](https://github.com/iotaledger/stronghold.rs/blob/snapshot-migration-v3age-2.0/client/src/procedures/primitives.rs#L587) and [`PublicKey`](https://github.com/iotaledger/stronghold.rs/blob/snapshot-migration-v3age-2.0/client/src/procedures/primitives.rs#L612) procedures). We can make `hazmat` module public in iota-crypto-0.20.1, but it would be difficult to completely fix stronghold.